### ISSUE TITLE
Unintended game-breaking issue with tutorial "Expand the Town Map Tileset"

### ISSUE THREAD

**Issue by Maesejesus** [pret/pokecrystal#721]
I'm sorry again if this is not the best place to post this, but what I'm going to mention is not an issue with Pokecrystal itself, but with one of the tutorials, and I have no idea where to begin to fix it so I can't really write the solution of this issue myself like with the Bag Pocket time ago yet I still to warn others about this. 

The "Expand the Town Map Tileset" works well with what it tries to do... But doing so introduces a rather big glitch: If a Pokémon has Mail, and you look at your map (our you use Fly which brings the map up) it corrupts the Mail.

Once you look at the map once, every Mail in your party becomes a glitchfest. All the data screws up (type of mail, content, sender, etc) and if you try to check it the effects are varied, from simply corrupting the graphics, to showing an empty screen, to basically freezing the game in different ways.

The problem starts somewhere on step 2 "Move the "extra" font graphics and use their space". This glitch does not happen anywhere during the previous steps of this (including the "Expand tilesets from 192 to 255" tutorial).

Again, I apologize for opening an issue not strictly related with the disassembly itself.

**Comment by Idain** [comment]
I'll check this later when I get home. This sounds interesting.

**Comment by Caverns4** [comment]
I was wondering why the mail was getting corrupted in my modification... Good to know what the cause is.

**Comment by coltongit** [comment]
I've tweaked the tutorials so they link to each other and touches on the point this issue brings up

**Comment by coltongit** [comment]
> Well, this was two years ago, so it's a bit hazy by now, but I ended doing something else to fix my issues, creating a second version of "LoadTownMapGFX" (just named it "LoadTownMapGFX2" myself) so it can take charge of the code for Fly and Pokédex Area maps, and leaving the original for the PokeGear map use.
> 
> The normal LoadTownMapGFX was left with "BANK(TownMapGFX), 50" while the second had "BANK(TownMapGFX), 96". It has been working fine since then.
> 
> Going back to what the proposed tweak says, having only one LoadTownMapGFX with "BANK(TownMapGFX), 150" seemingly makes all maps work fine and the mail doesn't get corrupted but it, at least on my project, corrupts the text area on the upper side of the screen and after that the other pages of the PokeGear get also corrupted.
> 
> But again, might be just on my end with my existing hack. I kindly invite you to try what I did.
> 
> LoadTownMapGFX: ld hl, TownMapGFX ld de, vTiles2 lb bc, BANK(TownMapGFX), 50 call DecompressRequest2bpp ret
> 
> LoadTownMapGFX2: ld hl, TownMapGFX ld de, vTiles2 lb bc, BANK(TownMapGFX), 96 call DecompressRequest2bpp ret
> 
> Then change the references of LoadTownMapGFX to LoadTownMapGFX2 on _FlyMap: and Pokedex_GetArea:

perhaps if the tweak from #2 was applied to #1 it'll work that way too, I'll tweak the wiki page again so you can test on a clean pokecrystal

**Comment by Maesejesus** [comment]
If I change TownMapGFX from 50 to 96 glitches come back to the maps, the mail fortunately doesn't break. I remember bashing my head against this back in the day and the way I ended up doing it was the only that worked fine for me. 

Whenever I get time I'll try on a clean pokecrystal, but seeing how other people had the same issue there might not be much of a difference between a clean one or my hack in this particular matter.

**Comment by coltongit** [comment]
> If I change TownMapGFX from 50 to 96 glitches come back to the maps, the mail fortunately doesn't break. I remember bashing my head against this back in the day and the way I ended up doing it was the only that worked fine for me.
> 
> Whenever I get time I'll try on a clean pokecrystal, but seeing how other people had the same issue there might not be much of a difference between a clean one or my hack in this particular matter.

Perhaps I should be doing it the way you suggested proper regardless if the 192->255 tiles tutorial was done or not? creating that second function and updating the references to it in the Pokégear so it doesn't corrupt anything?

**Comment by Maesejesus** [comment]
No, it does not corrupt anything, every screen looks fine and there's no corrupted tiles while going through screens when having two TownMapGFX. And I added the 192>255 Tiles expansion previously, like the tutorial said back in the day. 
![pokecrystal11 fixed 2](https://user-images.githubusercontent.com/21066817/175286454-5d3047af-3a8c-4994-b2ad-7b127bf68081.png)
![pokecrystal11 Fixed](https://user-images.githubusercontent.com/21066817/175286458-bd4f5ea9-7c5d-4d40-a1e6-3bf0ab9ac314.png)

These are from my hack with the way I did it. Everything works on any Map screen, be it Pokédex, Fly, PokeGear or the Map you can find in some houses.

This is with "lb bc, BANK(TownMapGFX), 96":
Fly and Pokédex Maps work, the Pokegear misses some graphics such as the Tuning dial ones:

![pokecrystal11 96 2](https://user-images.githubusercontent.com/21066817/175287233-31763672-6c5c-461c-a8e5-42a5743c52ce.png)
![pokecrystal11 96 1](https://user-images.githubusercontent.com/21066817/175287237-15a7c2d6-898f-471e-833d-5b30ad4f00ab.png)


With the proposed early "150" it got these glitches:

![pokecrystal11 150 2](https://user-images.githubusercontent.com/21066817/175287508-79e52240-85c3-4b55-be26-68e25838200f.png)
![pokecrystal11 150 1](https://user-images.githubusercontent.com/21066817/175287514-067b072c-80e1-44dd-aaf6-0ef21cb0bd2d.png)

I'm faaar from being good at coding and I'm sure Rangi would find a better solution with closed eyes, but that's how I managed to avoid issues myself.

**Comment by coltongit** [comment]
> No, it does not corrupt anything, every screen looks fine and there's no corrupted tiles while going through screens when having two TownMapGFX. And I added the 192>255 Tiles expansion previously, like the tutorial said back in the day. ![pokecrystal11 fixed 2](https://user-images.githubusercontent.com/21066817/175286454-5d3047af-3a8c-4994-b2ad-7b127bf68081.png) ![pokecrystal11 Fixed](https://user-images.githubusercontent.com/21066817/175286458-bd4f5ea9-7c5d-4d40-a1e6-3bf0ab9ac314.png)
> 
> These are from my hack with the way I did it. Everything works on any Map screen, be it Pokédex, Fly, PokeGear or the Map you can find in some houses.
> 
> This is with "lb bc, BANK(TownMapGFX), 96": Fly and Pokédex Maps work, the Pokegear misses some graphics such as the Tuning dial ones:
> 
> ![pokecrystal11 96 2](https://user-images.githubusercontent.com/21066817/175287233-31763672-6c5c-461c-a8e5-42a5743c52ce.png) ![pokecrystal11 96 1](https://user-images.githubusercontent.com/21066817/175287237-15a7c2d6-898f-471e-833d-5b30ad4f00ab.png)
> 
> With the proposed early "150" it got these glitches:
> 
> ![pokecrystal11 150 2](https://user-images.githubusercontent.com/21066817/175287508-79e52240-85c3-4b55-be26-68e25838200f.png) ![pokecrystal11 150 1](https://user-images.githubusercontent.com/21066817/175287514-067b072c-80e1-44dd-aaf6-0ef21cb0bd2d.png)
> 
> I'm faaar from being good at coding and I'm sure Rangi would find a better solution with closed eyes, but that's how I managed to avoid issues myself.

I added your fix how you showed it in https://github.com/pret/pokecrystal/issues/721#issuecomment-1164231737 and it should all be good now, with your confirmation here that it's fixed we can close this

**Comment by Idain** [comment]
> perhaps if the tweak from `LoadTownMapGFX2` was applied to `LoadTownMapGFX` it'll work that way too, I'll tweak the wiki page again so you can test on a clean pokecrystal EDIT: Wiki page tweaked with your fix, give it another shot so I know I didn't break anything again

Try not to post untested fixes on the wiki page, @coltongit. If someone unexperienced sees one of them they'll blindly implement it and then complain that it doens't work.

**Comment by vulcandth** [comment]
@coltongit Did you test any of these changes before modifying the wiki page? Do **you** understand the problem and **why** those changes would fix that problem? Or are you simply trusting that Maesejesus gave you a proper fix? No offense to Maesejesus, but they didn't seem wholly confident their solution was the correct one. Blindly fixing problems, without understanding them is a bad idea; as you may cause other issues. The original issue to this problem was the corruption of mail, that went unnoticed for a while. How do you know that something else wasn't corrupted by applying this "fix"?
